### PR TITLE
fix: test regression since #4920

### PR DESCRIPTION
**Pull request by harshavardhana**
## Community Contribution License
All community contributions in this pull request are licensed to the project maintainers
under the terms of the [Apache 2 license](https://www.apache.org/licenses/LICENSE-2.0). 
By creating this pull request I represent that I have the right to license the 
contributions to the project maintainers under the Apache 2 license.

## Description
fix: test regression since #4920

## Motivation and Context
use mc_cmd always inside `functional-tests.sh`

Bonus: run shell formatting

Fix breaking builds on MinIO https://github.com/minio/minio/actions/runs/9015499066/job/24770248936?pr=19705

## How to test this PR?
CI/CD is failing on MinIO repo

## Types of changes
- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Optimization (provides speedup with no functional changes)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:
- [x] Fixes a regression in tests since #4920
- [ ] Unit tests added/updated
- [ ] Internal documentation updated
- [ ] Create a documentation update request [here](https://github.com/minio/docs/issues/new?label=doc-change,title=Doc+Updated+Needed+For+PR+github.com%2fminio%2fmc%2fpull%2fNNNNN)

### CODE REVIEW

File: functional-tests.sh [hunk 1]
```diff
@@ -96,1139 +96,1083 @@ MC_CONFIG_DIR="/tmp/.mc-$RANDOM"
...
+	assert_success "$start_time" "${FUNCNAME[0]}" mc_cmd version enable "${SERVER_ALIAS}/${bkt1}"
+	assert_success "$start_time" "${FUNCNAME[0]}" mc_cmd mb "${SERVER_ALIAS}/${bkt2}"
+	assert_success "$start_time" "${FUNCNAME[0]}" mc_cmd version enable "${SERVER_ALIAS}/${bkt2}"
+	assert_success "$start_time" "${FUNCNAME[0]}" mc_cmd replicate add "${SERVER_ALIAS}/${bkt1}" --remote-bucket "${SERVER_ALIAS}/${bkt2}"
+	loop_count=0
+	while true; do
+		if [ $loop_count -eq 100 ]; then
+			break
+		fi
+		# Intentionally not using mc_cmd to avoid delays of adding error to log files etc
```
Comment (shtripat) [inline functional-tests.sh:1042]:
```suggestion
```

File: functional-tests.sh [hunk 2]
```diff
@@ -96,1139 +96,1083 @@ MC_CONFIG_DIR="/tmp/.mc-$RANDOM"
...
+	object_name="mc-test-replicate-$RANDOM"
+	bkt1="rep-bucket1"
+	bkt2="rep-bucket2"
+
+	echo "${SERVER_ALIAS}"
+	mc_cmd admin info "${SERVER_ALIAS}"
+	mc_cmd ls "${SERVER_ALIAS}"
+	# Remove any pre-existing buckets
+	mc_cmd rb "${SERVER_ALIAS}/${bkt1}" --force
+	mc_cmd rb "${SERVER_ALIAS}/${bkt2}" --force
```
Comment (shtripat) [inline functional-tests.sh:1031]: Would these fail if we try removing non existing buckets?

**Review by shtripat — COMMENTED** [review]
couple of minor ones. else looks fine

File: functional-tests.sh (hunk 2 above)
Comment (harshavardhana) [reply functional-tests.sh:1031]: thats for a different PR to deal with :-) @shtripat

**Review by shtripat — APPROVED** [review]
lgtm

**Pull request merged**